tests/jballerina-unit-test/src/test/resources/test-src/types/tuples/tuple_access_expr.bal

```diff
@@ -277,3 +277,152 @@ function testTupleAccessUsingUnionWithFiniteTypesNegative() {
     string|boolean f1 = tuple[index0];
     var f2 = tuple[index1];
 }
+
+type IntTuple [int...];
+type StringTuple [string...];
+type BooleanTuple [boolean...];
+type CustomTuple [int, string, boolean];
+
+const int index = 0;
+
+[int...] expectedIntTuple = [1, 2, 3];
+[string...] expectedStringTuple = ["string 1", "string 2", "string 3"];
+[boolean...] expectedBooleanTuple = [true, false, true];
+
+public function testTupleAccessWithCustomType() {
+    IntTuple intTuple = [1, 2, 3];
+    StringTuple stringTuple = ["string 1", "string 2", "string 3"];
+    BooleanTuple booleanTuple = [true, false, true];
+    CustomTuple customTuple = [1, "string 1", true];
+
+    assertEquals(intTuple[index], 1);
+    assertEquals(intTuple[index + 1], 2);
+    assertEquals(intTuple[index + 2], 3);
+
+    assertEquals(stringTuple[index], "string 1");
+    assertEquals(stringTuple[index + 1], "string 2");
+    assertEquals(stringTuple[index + 2], "string 3");
+
+    assertEquals(booleanTuple[index], true);
+    assertEquals(booleanTuple[index + 1], false);
+    assertEquals(booleanTuple[index + 2], true);
+
+    assertEquals(customTuple[index], 1);
+    assertEquals(customTuple[index + 1], "string 1");
+    assertEquals(customTuple[index + 2], true);
+
+    foreach var i in 0 ..< intTuple.length() {
+        assertEquals(intTuple[i], expectedIntTuple[i]);
+        assertEquals(stringTuple[i], expectedStringTuple[i]);
+        assertEquals(booleanTuple[i], expectedBooleanTuple[i]);
```

Comment:
I would rather assert the access on the module-level and local variables separately since this adds a dependency between the two variables.

Reply:
Updated. 0b97792c352b9e0d5bb788ace062d3713de2d277